server/src/main/java/org/eclipse/openvsx/migration/OrphanNamespaceMigration.java

```diff
@@ -0,0 +1,85 @@
+/********************************************************************************
+ * Copyright (c) 2020 TypeFox and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ********************************************************************************/
+package org.eclipse.openvsx.migration;
+
+import java.util.LinkedHashSet;
+
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+
+import org.eclipse.openvsx.entities.NamespaceMembership;
+import org.eclipse.openvsx.entities.UserData;
+import org.eclipse.openvsx.repositories.RepositoryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationStartedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrphanNamespaceMigration {
+
+    protected final Logger logger = LoggerFactory.getLogger(OrphanNamespaceMigration.class);
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    RepositoryService repositories;
+
+    @EventListener
+    @Transactional
+    public void fixOrphanNamespaces(ApplicationStartedEvent event) {
+        int[] count = new int[3];
+        repositories.findOrphanNamespaces().forEach(namespace -> {
+            var extensions = repositories.findExtensions(namespace);
+            if (extensions.isEmpty()) {
+                // The namespace is orphaned and empty - remove it
+                entityManager.remove(namespace);
+                count[0]++;
+            } else {
+
+                // Find all previous contributors
+                var contributors = new LinkedHashSet<UserData>();
+                for (var extension : extensions) {
+                    for (var extVersion : repositories.findActiveVersions(extension)) {
```

Comment:
why is it important to differentiate between those extensions for that matter? is an inactive extension not linked to a publisher/user?

Reply:
Currently the only reasons for an extension to be inactive are
 * the publisher has not signed the [Publisher Agreement](https://open-vsx.org/publisher-agreement-v1.0) after the grace period has elapsed (January 9, 2021),
 * the publisher expresses the wish to revoke the Publisher Agreement.
 
My thought here was that we shouldn't automatically assign memberships to such users.